### PR TITLE
Treat all projects in LUNA_LIBS_PATH as available to import

### PR DESCRIPTION
**Pull request by mikusp**
### Pull Request Description

This pull request causes Luna to treat all projects located at LUNA_LIBS_PATH as available to import. It was done to prepare a package with Dataframes embedded and seems to work as intended on Windows.

### Important Notes

Projects discovery is rather simple, treating all directories starting with capital letter as Luna projects.

### Checklist
Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [ ] The code conforms to the [Luna Style Guide](https://github.com/luna/wiki/blob/master/code-style/01.general.md).
- [ ] The code has been tested where possible.

### CODE REVIEW

File: package/src/Luna/Package.hs [hunk 1]
```diff
@@ -207,26 +209,35 @@ listDependencies pkgSrc = Exception.rethrowFromIO @Path.PathException $ do
...
+        =<< Environment.getEnv Name.lunaRootEnv
+    projectNames <- liftIO $ do
+        contents <- Directory.listDirectory lunaroot
+        dirs     <- filterM
+            (\a -> Directory.doesDirectoryExist $ lunaroot FilePath.</> a) contents
+        let projects = filter
+                (\a -> (isUpper <$> head a) == Just True) dirs
+        return projects
+    for projectNames $ \projectName ->
+        pure (convert projectName, lunaroot <> "/" <> projectName <> "/")
```
Comment (iamrecursion) [inline package/src/Luna/Package.hs]: This is non-portable to windows due to the hardcoded `"/"`. You need to use either `</>` or `pathSeparator` from `System.FilePath`.

Comment (mikusp) [reply package/src/Luna/Package.hs]: It was tested on Windows and worked fine

Comment (iamrecursion) [reply package/src/Luna/Package.hs]: Interesting! I'd still prefer use of `pathSeparator`, but if it works that's purely a cosmetic nitpick. Approved!

Comment (mwu-tow) [reply package/src/Luna/Package.hs]: Windows itself (as in NT kernel) accepts both `\` and `/`, you can even freely mix them together. However many applications (including cmd and explorer) assume backslashes as separators and don't fully support forward slashes. It is strongly recommended to use `\` as path separator for consistency and so path eg. can be easily pasted into explorer.

**Comment by piotrMocz** [comment]
@mikusp I think this is good to be merged, am I right?

**Comment by mikusp** [comment]
It's good and working but not tested.

pon., 8 paź 2018, 22:52 użytkownik piotrMocz <notifications@github.com>
napisał:

**Pull request merged**